src/Button/Button.js

```diff
@@ -37,18 +41,85 @@ const Button = React.forwardRef(({
         className
     );
 
-    return (<button {...props} className={buttonClasses}
-        disabled={disabled} onClick={onClick}
-        ref={ref}
-        selected={selected}
-        type={typeAttr}>
-        {children}
-    </button>);
+    const ariaDisabled = props['aria-disabled'];
+    let disabledInAnyWay = disabled || ariaDisabled;
+
+    //if button is disabled and focusable, remove onClick handler
+    let onClickHandler = allowFocusOnDisable && disabledInAnyWay ? () => {} : onClick;
+
+    const ariaLiveId = shortId.generate();
+
+    const getDisabledProps = () => {
+        let disabledProps = {};
+        if (allowFocusOnDisable) {
+            disabledProps['aria-disabled'] = disabledInAnyWay;
+            disabledProps['aria-describedby'] = ariaLiveId;
+        } else {
+            disabledProps.disabled = disabledInAnyWay;
+        }
+        return disabledProps;
+    };
+
+    const liveMessageForScreenReaders = disabledInAnyWay ? disabledMessage : enabledMessage;
+
+    const renderButtonStateMessage = () => {
+        let content = '';
```

Comment:
Small nit: this seems to add a new line on every storyshot for every component that uses a button, probably due to the empty string `content`. I wonder if this should be instead be `null`.

Reply:
Oh holy moly, I was wondering why there were a bunch of blank lines when I updated the story snapshots. Yeah I agree this should be null. I will update and test to make sure there are no adverse effects. Thanks!